pydmd/lando.py

```diff
@@ -0,0 +1,1089 @@
+        self._kernel_params = kernel_params
+        self._kernel_function = kernel_function
+        self._kernel_gradient = kernel_gradient
+
+        # Build the LANDO operator.
+        self._Atilde = LANDOOperator(
+            svd_rank=svd_rank,
+            dict_tol=dict_tol,
+            online=online,
+            permute=permute,
+            lstsq=lstsq,
+        )
+
+        # Keep track of the data rescaling factor.
+        self._x_rescale = x_rescale
+
+        # Keep track of the last fixed point analysis.
+        self._compute_A = False
+        self._fixed_point = None
+        self._bias = None
+
+        # Keep track of whether or not the model was ever updated.
+        self._updated = False
+
+    def kernel_function(self, X, Y):
+        """
+        Calls `sklearn.metrics.pairwise_kernels` to evaluate the kernel
+        function at the given feature matrices X and Y. If a custom kernel
+        function was provided, then that function will be used instead.
+
+        :param X: feature matrix with shape (n_features, n_samples_X)
+        :type X: numpy.ndarray
+        :param Y: second feature matrix with shape (n_features, n_samples_Y)
+        :type Y: numpy.ndarray
+        :return: kernel matrix with shape (n_samples_X, n_samples_Y)
+        :rtype: numpy.ndarray
+        """
+        if self._kernel_function is not None:
+            return self._kernel_function(X, Y)
+        return pairwise_kernels(
+            X.T, Y.T, metric=self._kernel_metric, **self._kernel_params
+        )
+
+    def kernel_gradient(self, X, y):
+        """
+        Computes the gradient of the kernel with respect to the second feature
+        vector x. Then evaluates the gradient at the feature matrix X and the
+        feature vector y. Currently, this method is only compatible with
+        the linear, polynomial, and RBF kernels, unless a custom kernel
+        gradient function was provided.
+
+        :param X: feature matrix with shape (n_features, n_samples_X)
+        :type X: numpy.ndarray
+        :param y: feature vector with shape (n_features,)
+        :type y: numpy.ndarray
+        :return: kernel gradient matrix with shape (n_samples_X, n_features)
+        :rtype: numpy.ndarray
+        """
+        if self._kernel_function is not None:
+            if self._kernel_gradient is None:
+                msg = (
+                    "Unable to evaluate the gradient of the kernel function. "
+                    "If a custom kernel function was provided, please ensure "
+                    "that the corresponding gradient function is also defined "
+                    "by setting the kernel_gradient parameter the LANDO model."
+                )
+                raise ValueError(msg)
+            return self._kernel_gradient(X, y)
+
+        if self._kernel_metric == "linear":
+            return X.T
+
+        # Kernel metric must be polynomial or RBF.
+        if "gamma" in self._kernel_params.keys():
```

Comment:
`.keys()` is not needed:
```suggestion
        if "gamma" in self._kernel_params:
```

Reply:
Same for other occurrences below

Reply:
Fixed!